gossip/hub/handlers.go

```diff
@@ -626,6 +630,17 @@ func chargeTo(ctx context.Context) *trillian.ChargeTo {
 	return &trillian.ChargeTo{User: []string{qUser}}
 }
 
+// appendCharge adds the specified user to the passed in ChargeTo and and
```

Comment:
Modifying `a` feels like a layering violation. What is the intended use case? 
In the context of `chargeTo(ctx)`, modifying `a` won't store the value back in `ctx`. 
I worry that this is non-obvious for callers.

Reply:
I don't really understand your comment -- there's no layers here to violate.  This function does what it says on the tin - adds an extra `user` into a `ChargeTo`, so the use case is any code that wants to do that.

(That being said, it's usually nicer to return a new object rather than modifying one that's passed in, so I've changed that, as per the other suggestion.)

The particular use at l.306 adds a sourceID `user` into a `ChargeTo` created from a context.  There's no need or attempt to store anything back into a context.